parser/src/main/java/com/facebook/coresql/warning/CoreSqlWarning.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.warning;
+
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class CoreSqlWarning
+{
+    private final WarningCode warningCode;
+    private final String message;
+
+    public CoreSqlWarning(WarningCode standardWarningCode, String message)
+    {
+        this.warningCode = requireNonNull(standardWarningCode, "warningCode is null");
+        this.message = requireNonNull(message, "message is null");
+    }
+
+    public CoreSqlWarning(WarningCodeSupplier warningCodeSupplier, String message)
+    {
+        this(requireNonNull(warningCodeSupplier, "warningCodeSupplier is null").toWarningCode(), message);
+    }
+
+    public WarningCode getWarningCode()
+    {
+        return warningCode;
+    }
+
+    public String getMessage()
+    {
+        return message;
+    }
+
+    @Override
+    public boolean equals(Object o)
```

Comment:
Do we really need equals and hashcode?

Reply:
Probably not. I'll remove it and we can add it back in if necessary.